PSReadLine/DynamicHelp.cs

```diff
@@ -0,0 +1,289 @@
+        }
+
+        object IPSConsoleReadLineMockableMethods.GetDynamicHelpContent(string commandName, string parameterName, bool isFullHelp)
+        {
+            if (!String.IsNullOrEmpty(commandName) && isFullHelp)
+            {
+                _ps ??= System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace);
+                _ps.Commands.Clear();
+
+                return _ps
+                    .AddCommand($"Microsoft.PowerShell.Core\\Get-Help")
+                    .AddParameter("Name", commandName)
+                    .AddParameter("Full", value: true)
+                    .AddCommand($"Microsoft.PowerShell.Utility\\Out-String")
+                    .Invoke<string>()
+                    .FirstOrDefault();
+            }
+            else if (!String.IsNullOrEmpty(commandName) && !String.IsNullOrEmpty(parameterName) && !isFullHelp)
+            {
+                _ps ??= System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace);
+                _ps.Commands.Clear();
+
+                return _ps
+                    .AddCommand("Microsoft.PowerShell.Core\\Get-Help")
+                    .AddParameter("Name", commandName)
+                    .AddParameter("Parameter", parameterName)
+                    .Invoke<PSObject>()
+                    .FirstOrDefault();
+            }
+
+            return null;
+        }
+
+        void IPSConsoleReadLineMockableMethods.WriteToPager(string content, string regexPatternToScrollTo)
+        {
+            _pager.Write(content, regexPatternToScrollTo);
+        }
+
+        private void WriteDynamicHelpContent(string commandName, string parameterName, bool isFullHelp)
+        {
+            var helpContent = _mockableMethods.GetDynamicHelpContent(commandName, parameterName, isFullHelp);
+
+            if (!String.IsNullOrEmpty(commandName) && isFullHelp)
+            {
+                if (helpContent is String fullHelp && !String.IsNullOrEmpty(fullHelp))
+                {
+                    string regexPatternToScrollTo = null;
+
+                    if (!String.IsNullOrEmpty(parameterName))
+                    {
+                        string upper = parameterName[0].ToString().ToUpperInvariant();
+                        string lower = parameterName[0].ToString().ToLowerInvariant();
+                        string remainingString = parameterName.Substring(1);
+                        regexPatternToScrollTo = $"-[{upper}|{lower}]{remainingString} [<|\\[]";
+                    }
+
+                    _mockableMethods.WriteToPager(fullHelp, regexPatternToScrollTo);
+                }
+            }
+            else if (!String.IsNullOrEmpty(commandName) && !String.IsNullOrEmpty(parameterName))
+            {
+                if (helpContent is PSObject paramHelp)
+                {
+                    WriteParameterHelp(paramHelp);
+                }
+            }
+        }
+
+        private void DynamicHelpImpl(bool isFullHelp)
```

Comment:
Full help won't work on old Windows OS where the console doesn't support ANSI sequences, how are you handling that?
Parameter help should work, but can you please validate?